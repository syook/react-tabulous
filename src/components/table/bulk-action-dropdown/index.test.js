import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import BulkActionList from '.';

configure({ adapter: new Adapter() });

describe('BulkActionList', () => {
  it('should render without crashing', () => {
    shallow(<BulkActionList />);
  });

  it('should match snapshot', () => {
    const tree = mount(<BulkActionList />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should contain Dropdown component', () => {
    const wrapper = mount(<BulkActionList />);
    expect(wrapper.find('Dropdown').exists()).toBe(true);
  });

  it('Dropdown text should be selectedRows.length', () => {
    const wrapper = mount(<BulkActionList selectedRows={[1, 2]} />);
    expect(wrapper.find('Dropdown').props().text).toBe('Bulk Action (2 selected)');
  });

  it('Dropdown text should be zero if selectedRows are empty', () => {
    const wrapper = mount(<BulkActionList selectedRows={[]} />);
    expect(wrapper.find('Dropdown').props().text).toBe('Bulk Action (0 selected)');
  });

  it('should match snapshot is selectedRows are empty', () => {
    const tree = mount(<BulkActionList selectedRows={[]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should match snapshot is selectedRows are not empty', () => {
    const tree = mount(<BulkActionList selectedRows={[1, 2]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should match snapshot if bulkActions are empty', () => {
    const tree = mount(<BulkActionList selectedRows={[1, 2]} bulkActions={[]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should match snapshot if bulkActions are not empty', () => {
    const bulkActions = [{ function: () => {}, name: 'Delete' }];
    const tree = mount(<BulkActionList selectedRows={[1, 2]} bulkActions={bulkActions} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
