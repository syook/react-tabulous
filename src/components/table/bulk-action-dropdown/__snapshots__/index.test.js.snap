// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkActionList should match snapshot 1`] = `
<BulkActionList>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    button={true}
    className="icon bulk-action right labeled"
    closeOnBlur={true}
    deburr={false}
    icon="dropdown"
    minCharacters={1}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    searchInput="text"
    selectOnBlur={true}
    selectOnNavigation={true}
    text="Bulk Action (0 selected)"
    wrapSelection={true}
  >
    <Ref
      innerRef={
        Object {
          "current": <div
            aria-expanded="false"
            class="ui button dropdown icon bulk-action right labeled"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-live="polite"
              class="text"
              role="alert"
            >
              Bulk Action (0 selected)
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            />
          </div>,
        }
      }
    >
      <RefFindNode
        innerRef={
          Object {
            "current": <div
              aria-expanded="false"
              class="ui button dropdown icon bulk-action right labeled"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-live="polite"
                class="text"
                role="alert"
              >
                Bulk Action (0 selected)
              </div>
              <i
                aria-hidden="true"
                class="dropdown icon"
              />
              <div
                class="menu transition"
              />
            </div>,
          }
        }
      >
        <div
          aria-expanded={false}
          className="ui button dropdown icon bulk-action right labeled"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          >
            Bulk Action (0 selected)
          </div>
          <Icon
            as="i"
            className=""
            name="dropdown"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
          </Icon>
          <DropdownMenu
            className=""
          >
            <div
              className="menu transition"
            />
          </DropdownMenu>
        </div>
      </RefFindNode>
    </Ref>
  </Dropdown>
</BulkActionList>
`;

exports[`BulkActionList should match snapshot if bulkActions are empty 1`] = `
<BulkActionList
  bulkActions={Array []}
  selectedRows={
    Array [
      1,
      2,
    ]
  }
>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    button={true}
    className="icon bulk-action right labeled"
    closeOnBlur={true}
    deburr={false}
    icon="dropdown"
    minCharacters={1}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    searchInput="text"
    selectOnBlur={true}
    selectOnNavigation={true}
    text="Bulk Action (2 selected)"
    wrapSelection={true}
  >
    <Ref
      innerRef={
        Object {
          "current": <div
            aria-expanded="false"
            class="ui button dropdown icon bulk-action right labeled"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-live="polite"
              class="text"
              role="alert"
            >
              Bulk Action (2 selected)
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            />
          </div>,
        }
      }
    >
      <RefFindNode
        innerRef={
          Object {
            "current": <div
              aria-expanded="false"
              class="ui button dropdown icon bulk-action right labeled"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-live="polite"
                class="text"
                role="alert"
              >
                Bulk Action (2 selected)
              </div>
              <i
                aria-hidden="true"
                class="dropdown icon"
              />
              <div
                class="menu transition"
              />
            </div>,
          }
        }
      >
        <div
          aria-expanded={false}
          className="ui button dropdown icon bulk-action right labeled"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          >
            Bulk Action (2 selected)
          </div>
          <Icon
            as="i"
            className=""
            name="dropdown"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
          </Icon>
          <DropdownMenu
            className=""
          >
            <div
              className="menu transition"
            />
          </DropdownMenu>
        </div>
      </RefFindNode>
    </Ref>
  </Dropdown>
</BulkActionList>
`;

exports[`BulkActionList should match snapshot if bulkActions are not empty 1`] = `
<BulkActionList
  bulkActions={
    Array [
      Object {
        "function": [Function],
        "name": "Delete",
      },
    ]
  }
  selectedRows={
    Array [
      1,
      2,
    ]
  }
>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    button={true}
    className="icon bulk-action right labeled"
    closeOnBlur={true}
    deburr={false}
    icon="dropdown"
    minCharacters={1}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    searchInput="text"
    selectOnBlur={true}
    selectOnNavigation={true}
    text="Bulk Action (2 selected)"
    wrapSelection={true}
  >
    <Ref
      innerRef={
        Object {
          "current": <div
            aria-expanded="false"
            class="ui button dropdown icon bulk-action right labeled"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-live="polite"
              class="text"
              role="alert"
            >
              Bulk Action (2 selected)
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            >
              <div
                class="item"
                role="option"
              >
                <span
                  class="text"
                >
                  Delete
                </span>
              </div>
            </div>
          </div>,
        }
      }
    >
      <RefFindNode
        innerRef={
          Object {
            "current": <div
              aria-expanded="false"
              class="ui button dropdown icon bulk-action right labeled"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-live="polite"
                class="text"
                role="alert"
              >
                Bulk Action (2 selected)
              </div>
              <i
                aria-hidden="true"
                class="dropdown icon"
              />
              <div
                class="menu transition"
              >
                <div
                  class="item"
                  role="option"
                >
                  <span
                    class="text"
                  >
                    Delete
                  </span>
                </div>
              </div>
            </div>,
          }
        }
      >
        <div
          aria-expanded={false}
          className="ui button dropdown icon bulk-action right labeled"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          >
            Bulk Action (2 selected)
          </div>
          <Icon
            as="i"
            className=""
            name="dropdown"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
          </Icon>
          <DropdownMenu
            className=""
          >
            <div
              className="menu transition"
            >
              <DropdownItem
                key="BulkActionList-0"
                onClick={[Function]}
                text="Delete"
              >
                <div
                  className="item"
                  onClick={[Function]}
                  role="option"
                >
                  <span
                    className="text"
                  >
                    Delete
                  </span>
                </div>
              </DropdownItem>
            </div>
          </DropdownMenu>
        </div>
      </RefFindNode>
    </Ref>
  </Dropdown>
</BulkActionList>
`;

exports[`BulkActionList should match snapshot is selectedRows are empty 1`] = `
<BulkActionList
  selectedRows={Array []}
>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    button={true}
    className="icon bulk-action right labeled"
    closeOnBlur={true}
    deburr={false}
    icon="dropdown"
    minCharacters={1}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    searchInput="text"
    selectOnBlur={true}
    selectOnNavigation={true}
    text="Bulk Action (0 selected)"
    wrapSelection={true}
  >
    <Ref
      innerRef={
        Object {
          "current": <div
            aria-expanded="false"
            class="ui button dropdown icon bulk-action right labeled"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-live="polite"
              class="text"
              role="alert"
            >
              Bulk Action (0 selected)
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            />
          </div>,
        }
      }
    >
      <RefFindNode
        innerRef={
          Object {
            "current": <div
              aria-expanded="false"
              class="ui button dropdown icon bulk-action right labeled"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-live="polite"
                class="text"
                role="alert"
              >
                Bulk Action (0 selected)
              </div>
              <i
                aria-hidden="true"
                class="dropdown icon"
              />
              <div
                class="menu transition"
              />
            </div>,
          }
        }
      >
        <div
          aria-expanded={false}
          className="ui button dropdown icon bulk-action right labeled"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          >
            Bulk Action (0 selected)
          </div>
          <Icon
            as="i"
            className=""
            name="dropdown"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
          </Icon>
          <DropdownMenu
            className=""
          >
            <div
              className="menu transition"
            />
          </DropdownMenu>
        </div>
      </RefFindNode>
    </Ref>
  </Dropdown>
</BulkActionList>
`;

exports[`BulkActionList should match snapshot is selectedRows are not empty 1`] = `
<BulkActionList
  selectedRows={
    Array [
      1,
      2,
    ]
  }
>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    button={true}
    className="icon bulk-action right labeled"
    closeOnBlur={true}
    deburr={false}
    icon="dropdown"
    minCharacters={1}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    searchInput="text"
    selectOnBlur={true}
    selectOnNavigation={true}
    text="Bulk Action (2 selected)"
    wrapSelection={true}
  >
    <Ref
      innerRef={
        Object {
          "current": <div
            aria-expanded="false"
            class="ui button dropdown icon bulk-action right labeled"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-live="polite"
              class="text"
              role="alert"
            >
              Bulk Action (2 selected)
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            />
          </div>,
        }
      }
    >
      <RefFindNode
        innerRef={
          Object {
            "current": <div
              aria-expanded="false"
              class="ui button dropdown icon bulk-action right labeled"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-live="polite"
                class="text"
                role="alert"
              >
                Bulk Action (2 selected)
              </div>
              <i
                aria-hidden="true"
                class="dropdown icon"
              />
              <div
                class="menu transition"
              />
            </div>,
          }
        }
      >
        <div
          aria-expanded={false}
          className="ui button dropdown icon bulk-action right labeled"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          >
            Bulk Action (2 selected)
          </div>
          <Icon
            as="i"
            className=""
            name="dropdown"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
          </Icon>
          <DropdownMenu
            className=""
          >
            <div
              className="menu transition"
            />
          </DropdownMenu>
        </div>
      </RefFindNode>
    </Ref>
  </Dropdown>
</BulkActionList>
`;
